src/fileformats/ocd_file_import.cpp

```diff
@@ -1084,42 +1085,32 @@ void OcdFileImport::setupBaseSymbol(Symbol* symbol, const OcdBaseSymbol& ocd_bas
 	symbol->setIsHelperSymbol(false);
 	symbol->setProtected(ocd_base_symbol.status & Ocd::SymbolProtected);
 	symbol->setHidden(ocd_base_symbol.status & Ocd::SymbolHidden);
-	
-	if (ocd_version == 8 && ocd_base_symbol.flags & 0x02)
+	setupIcon(symbol, ocd_base_symbol);
+}
+
+
+template< >
+void OcdFileImport::setupIcon<Ocd::BaseSymbolV8>(Symbol* symbol, const Ocd::BaseSymbolV8& ocd_base_symbol)
+{
+	if (ocd_base_symbol.flags & 0x02)
 	{
-		auto warning = tr("In symbol symbol %1 '%2': %3").
-		               arg(symbol->getNumberAsString(), symbol->getName(),
-			               tr("Unsupported compressed icon"));
-		addWarning(warning);
-#ifdef MAPPER_DEVELOPMENT_BUILD
-		qDebug("%s", qPrintable(warning));
-		/// \todo Factor out generic hex dump function
-		constexpr auto extent = std::extent<typename std::remove_pointer<decltype(ocd_base_symbol.icon.bits)>::type>::value;
-		auto icon_bits = ocd_base_symbol.icon.bits;
-		for (int i = 0; i < 16; ++i)
+		try
 		{
-			QByteArray data;
-			data.reserve(80);
-			data.append(QByteArray("000")
-			            .append(QByteArray::number(16*i,16))
-			            .right(4)
-			            .append("   "));
-			for (int j = 0; j < 17; ++j)
-			{
-				if (icon_bits == ocd_base_symbol.icon.bits + extent)
-					break;
-				if (*icon_bits < 0x10)
-					data += '0';
-				data += QByteArray::number(*(icon_bits++), 16) + ' ';
-			}
-			qDebug("%s", data.constData());
+			symbol->setCustomIcon(OcdIcon::toQImage(ocd_base_symbol.icon.uncompress()));
 		}
-#endif
-	}
-	else
-	{
-		symbol->setCustomIcon(OcdIcon::toQImage(ocd_base_symbol.icon));
+		catch (std::logic_error& e)
+		{
+			addWarning(tr(e.what()));
+		}
+		return;
 	}
+	symbol->setCustomIcon(OcdIcon::toQImage(ocd_base_symbol.icon));
+}
```

Comment:
If we want to extend the `try`-block, we shall make it a function-try-catch block.

The wiki is a little bit to rigid about exceptions IMO.
In the case of uncompress() they are purely exceptional indeed: They shouldn't ever happen except for data/programming errors. However, for compress(), it is more expected than exceptional to hit the limits of code table / output data size. Still, I would prefer to handle this with exceptions, in order to give the API a functional design, and because it is easy enough to deal with the unwinding.

Reply:
Ack.